test/test_common/simulated_time_system.cc

```diff
@@ -125,6 +143,32 @@ MonotonicTime SimulatedTimeSystem::monotonicTime() {
   return monotonic_time_;
 }
 
+void SimulatedTimeSystem::sleep(const Duration& duration) {
+  mutex_.lock();
+  MonotonicTime monotonic_time =
+      monotonic_time_ + std::chrono::duration_cast<MonotonicTime::duration>(duration);
+  setMonotonicTimeAndUnlock(monotonic_time);
+}
+
+Thread::CondVar::WaitStatus SimulatedTimeSystem::waitFor(Thread::MutexBasicLockable& mutex,
+                                                         Thread::CondVar& condvar,
+                                                         const Duration& duration) noexcept {
+  // Advance time to activate any callbacks. Note that activating them does not
+  // call them immediately; the libtimer event loop must be called, typically by
+  // Dispatcher::run() in a separate thread.
+  sleep(duration);
```

Comment:
Any concern with always advancing the clock? For instance if there are multiple timers that interact differently based on whether this method has returned yet or not. Perhaps it's worth a comment indicating that it will always full advance the timer.

Reply:
Actually that's a good point that had occurred to me before, and then I moved on. I'm glad you reminded me. I think it'd be better to advance only to the timestamp of the next alarm, and check if the condvar wakes up before trying again. That shouldn't be too hard.